Couple updates

godsgood33 opened the pull request:
I added some classes to make the library a little more robust.

godsgood33 committed:
Add new classes

godsgood33 committed:
fixed setter and added media data

godsgood33 committed:
Added .env

godsgood33 committed:
Added requirement for curl and dotenv libraries

godsgood33 committed:
Fixed license

godsgood33 committed:
Fixed size property and setter method

godsgood33 committed:
Fixed string output

godsgood33 committed:
Added media property

godsgood33 committed:
Tons of changes

jc21 commented:
Looks good :) Would you be able to add some more example usage to the readme for these new classes?

jc21 commented on src/jc21/Item.php:
```diff
@@ -0,0 +1,7 @@
+namespace jc21;
+
+class Item
+{
```
Could this class have something useful? maybe jsonSerialize?

godsgood33 replied:
The class is just intended to be a parent class so that you can put either the Movie or TV class into the same ItemCollection, otherwise you'd have to have 2 different collection classes.  I had intended on adding Photos and Music as options as well, but it appears that Plex treats those differently.
If you were to put jsonSerialize in the Item class it would only have access to properties with the Item class.  Now that I think about it, it may make more sense to make Item an interface so you could force compliance...I think this would still allow either class "implementing" the interface to be added to the collection.  I'll have to try that out.

godsgood33 commented:
> Looks good :) Would you be able to add some more example usage to the readme for these new classes?

Sure.  I wanted to see if you would interact with this before I started changing documentation.

godsgood33 committed:
Update documentation

godsgood33 committed:
Move files

godsgood33 committed:
Move symfony/dotenv to dev

godsgood33 committed:
Decided didn't need these classes

godsgood33 committed:
New utility docs

godsgood33 committed:
Doc updates

godsgood33 committed:
Moved files around for better understandability

godsgood33 commented:
Added some more doc and cleaned things up a bit.  I know there are more examples that could be made, but I have a trip coming up for a few days.  Wanted to give you something you could play around with a little.  I'm hoping to use this in a project I'm working on so I don't have to pass in the Plex library file path directly.

godsgood33 committed:
Don't actually need this

godsgood33 committed:
One too many lines

godsgood33 committed:
Added 'studio' as a valid filter

godsgood33 committed:
Added examples

godsgood33 committed:
Fixed minute method

godsgood33 committed:
Added url encoding to value and removed type

godsgood33 committed:
Removed gb, mb, kb measurements from string

godsgood33 committed:
Fix: single item returned

godsgood33 committed:
Add example

godsgood33 committed:
Added example for collection

godsgood33 commented:
Do you have any additional thoughts about these changes?

jc21 commented on composer.json:
```diff
@@ -12,14 +16,16 @@
     "minimum-stability": "dev",
     "require": {
-        "php": ">=5.3.0"
+        "php": ">=5.3.0",
```
Given the introduction of function return types and argument type declarations, this minimum version should probably be 7.4.0

godsgood33 committed:
Updated minimum PHP version to 7.4

godsgood33 commented:
Thanks for pushing and releasing.  The project I was referencing is [YTS-Parser](https://github.com/godsgood33/yts-parser)
